source/common/network/resolver_impl.cc

```diff
@@ -79,7 +80,9 @@ resolveProtoSocketAddress(const envoy::config::core::v3::SocketAddress& socket_a
   if (resolver == nullptr) {
     throwEnvoyExceptionOrPanic(fmt::format("Unknown address resolver: {}", resolver_name));
   }
-  return resolver->resolve(socket_address);
+  auto instance_or_error = resolver->resolve(socket_address);
+  THROW_IF_STATUS_NOT_OK(instance_or_error, throw);
+  return std::move(instance_or_error.value());
```

Comment:
Should we update the `resolveProtoAddress`  and `resolveProtoSocketAddress` to remove these `throwEnvoyExceptionOrPanic` and throw directly in this PR?

Reply:
largely I'm trying to tackle one concept at a time - I think I will get to them all (or at least the ones in the Envoy Mobile build, as we have compile checks on those)